hoomd/hpmc/pytest/test_faceted_ellipsoid.py

```diff
@@ -1,31 +1,314 @@
+                       (0, 1, 1),
+                       (1, 1, 0),
+                       (5, 0, 1),
+                       (0, 10, 1),
+                       (9, 5, 1),
+                       (0, 0, 1)],
+          "origin": (0, 0, 0),
+          "ignore_statistics": 1}
+
 
 def test_faceted_ellipsoid():
 
-    args_1 = {"normals": [(0, 0, 1), (0, 1, 0), (1, 0, 0), (0, 1, 1), (1, 1, 0), (1, 0, 1)], "offsets": [1, 3, 2, 6, 3, 1], "a": 3, "b": 4, "c": 1, "vertices": [(0, 0, 0), (0, 0, 1), (0, 1, 0), (1, 0, 0), (1, 1, 1), (1, 1, 0)], "origin": (0, 0, 0), "ignore_statistics":1}
-    test_faceted_ellipsoid1 = hpmc.faceted_ellipsoid_params(args_1)
+    test_faceted_ellipsoid1 = hpmc.FacetedEllipsoidParams(args_1)
     test_dict1 = test_faceted_ellipsoid1.asDict()
     assert test_dict1 == args_1
-    
-    args_2 = {"normals": [(0, 0, 0), (2, 1, 1), (1, 3, 3), (5, 1, 1), (1, 3, 0), (1, 2, 2)], "offsets": [1, 3, 3, 2, 3, 1], "a": 2, "b": 1, "c": 3, "vertices": [(1, 0, 0), (1, 1, 0), (1, 2, 1), (0, 1, 1), (1, 1, 2), (0, 0, 1)], "origin": (0, 0, 1), "ignore_statistics":0}
-    test_faceted_ellipsoid2 = hpmc.faceted_ellipsoid_params(args_2)
+
+    test_faceted_ellipsoid2 = hpmc.FacetedEllipsoidParams(args_2)
     test_dict2 = test_faceted_ellipsoid2.asDict()
     assert test_dict2 == args_2
-    
-    args_3 = {"normals": [(0, 0, 2), (0, 1, 1), (1, 3, 5), (0, 1, 6)], "offsets": [6, 2, 2, 5], "a": 1, "b": 6, "c": 6, "vertices": [(0, 0, 0), (1, 1, 1), (1, 0, 2), (2, 1, 1)], "origin": (0, 1, 0), "ignore_statistics":1}
-    test_faceted_ellipsoid3 = hpmc.faceted_ellipsoid_params(args_3)
+
+    test_faceted_ellipsoid3 = hpmc.FacetedEllipsoidParams(args_3)
     test_dict3 = test_faceted_ellipsoid3.asDict()
     assert test_dict3 == args_3
-    
-    args_4 = {"normals": [(0, 0, 2), (2, 2, 0), (3, 1, 1), (4, 1, 1), (1, 2, 0), (3, 3, 1), (1, 2, 1), (3, 3, 2)], "offsets": [5, 3, 3, 4, 3, 4, 2, 2], "a": 2, "b": 2, "c": 4, "vertices": [(0, 1, 0), (1, 1, 1), (1, 0, 1), (0, 1, 1), (1, 1, 0), (0, 0, 1), (0, 0, 1), (0, 0, 1)], "origin": (1, 0, 0), "ignore_statistics":0}
-    test_faceted_ellipsoid4 = hpmc.faceted_ellipsoid_params(args_4)
+
+    test_faceted_ellipsoid4 = hpmc.FacetedEllipsoidParams(args_4)
     test_dict4 = test_faceted_ellipsoid4.asDict()
     assert test_dict4 == args_4
 
-    args_5 = {"normals": [(0, 0, 1), (0, 4, 0), (2, 0, 1), (0, 3, 1), (4, 1, 0), (2, 2, 1), (1, 3, 1), (1, 9, 0), (2, 2, 2)], "offsets": [5, 4, 2, 2, 7, 3, 1, 4, 1], "a": 6, "b": 1, "c": 1, "vertices": [(0, 10, 3), (3, 2, 1), (1, 2, 1), (0, 1, 1), (1, 1, 0), (5, 0, 1), (0, 10, 1), (9, 5, 1), (0, 0, 1)], "origin": (0, 0, 0), "ignore_statistics":1}
-    test_faceted_ellipsoid5 = hpmc.faceted_ellipsoid_params(args_5)
+    test_faceted_ellipsoid5 = hpmc.FacetedEllipsoidParams(args_5)
     test_dict5 = test_faceted_ellipsoid5.asDict()
     assert test_dict5 == args_5
 
+
+def test_shape_params():
+
+    mc = hoomd.hpmc.integrate.FacetedEllipsoid(23456)
+
+    mc.shape['A'] = dict()
```

Comment:
Notice here how you iterate through all the keys here of by just enumerating them. You could do something like this
````
for key, value in mc.shape['A']:
     if key == 'ignore_statistics':
        assert value is False
    else:
       assert value == hoomd.typeconverter.RequiredArg
````
Also, `is` should work for `hoomd.typeconverter.RequiredArg` since is is a class that should only have one instance. This kind of pattern would work for all of these test. For instance, when filling out the type's shape using `args_{1,...,6}` you could check the full setting with a similar for loop. This would cover more, and reduce code (removing the potential for bugs). This applies to all the test files.